packages/runtime-core/__tests__/rendererElement.spec.ts

```diff
@@ -1,9 +1,55 @@
+import { nodeOps, render, h, serialize, Text } from '@vue/runtime-test'
+
 describe('renderer: element', () => {
-  test.todo('with props')
+  test('simple', () => {
+    const root = nodeOps.createElement('div')
+
+    render(h('div'), root)
+
+    expect(serialize(root)).toMatchSnapshot()
+  })
+
+  test('with props', () => {
+    const root = nodeOps.createElement('div')
+
+    render(
+      h('div', {
+        id: 'test'
+      }),
+      root
+    )
+
+    expect(serialize(root)).toMatchSnapshot()
+  })
+
+  test('with direct text children', () => {
+    const root = nodeOps.createElement('div')
+
+    render(h('div', ['I ❤ Vue']), root)
```

Comment:
This should be actual direct text, no array.
```suggestion
    render(h('div', 'I ❤ Vue'), root)
```